cmp/leftrightchooser/LeftRightChooserFilter.js

```diff
@@ -49,17 +51,21 @@ class LeftRightChooserFilter extends Component {
     }
     
     runFilter() {
-        const fields = this.props.fields || [],
+        const {fields = [], anyMatch} = this.props,
             searchTerm = escapeRegExp(this.value);
 
         const filter = (raw) => {
             return fields.some(f => {
                 const fieldVal = !!searchTerm && raw[f];
-                return ((fieldVal && new RegExp(`(^|\\\\W)${searchTerm}`, 'ig').test(fieldVal)) || !fieldVal);
+                return ((fieldVal && new RegExp(`${anyMatch ? '' : '(^|\\\\W)'}${searchTerm}`, 'ig').test(fieldVal)) || !fieldVal);
             });
         };
 
         this.model.setDisplayFilter(filter);
     }
+
+    componentWillUnmount() {
+        this.onClearClick();
```

Comment:
Good / interesting catch - assuming that without this a filter remained "stuck" in place at the model level - is that assumption correct?  (Just reading the code online here on Github at the moment.)

For clarity, I would rename the `onClearClick` handler as it is now now longer tightly bound to a clicking interaction.